### PR TITLE
Add basic Map with characters

### PR DESCRIPTION
**Pull request by calebhskim**

### CODE REVIEW

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      this.x = 0;
+      this.y = 0;
+   }
+
+   Bitmap createSubImageAt(final int row, final int col) {
+      // createBitmap(bitmap, x, y, width, height).
+      return Bitmap.createBitmap(image, col * objectWidth, row * objectHeight, objectWidth, objectHeight);
+   }
+
+   public int getX() {
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: this should be generated by lombok with the `@Getter` annotation

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      return this.x;
+   }
+
+   public int getY() {
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: this should be generated by lombok with the `@Getter` annotation

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      return this.y;
+   }
+
+   public int getHeight() {
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: this should be generated by lombok with the `@Getter` annotation

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      return objectHeight;
+   }
+
+   public int getWidth() {
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: this should be generated by lombok with the `@Getter` annotation

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
+package cpe307.team6.toweroffense.game;
+
+import android.graphics.Bitmap;
+
+public abstract class AbstractGameObject {
+   int objectWidth = 0;
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: this should be private

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+   int objectHeight = 0;
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: this should be private

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+   int x;
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: this should be private

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+   int y;
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: this should be private

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+   final int colCount;
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: this should be private

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+   final int rowCount;
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: this should be private

File: app/src/main/java/cpe307/team6/toweroffense/game/Explosion.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+   }
+
+   public void draw(final Canvas canvas) {
+      if (!finish) {
+         final Bitmap bitmap = this.createSubImageAt(rowIndex, colIndex);
+         canvas.drawBitmap(bitmap, this.x, this.y, null);
+      }
+   }
+
+   public boolean isFinish() {
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/Explosion.java]: this should be generated by lombok's `@Getter`

File: app/src/main/java/cpe307/team6/toweroffense/game/Explosion.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+      return finish;
+   }
+
+   public int getRowIndex() {
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/Explosion.java]: this should be generated by lombok's `@Getter`

File: app/src/main/java/cpe307/team6/toweroffense/game/Explosion.java [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+      return this.rowIndex;
+   }
+
+   public int getColIndex() {
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/Explosion.java]: this should be generated by lombok's `@Getter`

File: app/src/main/java/cpe307/team6/toweroffense/game/ChibiCharacter.java [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+      // Last draw time.
+      this.lastDrawNanoTime = System.nanoTime();
+   }
+
+   public void setMovingVector(final int movingVectorX, final int movingVectorY) {
+      ChibiCharacter.movingVectorX = movingVectorX;
+      ChibiCharacter.movingVectorY = movingVectorY;
+   }
+
+   public int getMovingVectorX() {
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/ChibiCharacter.java]: this should be generated by lombok's `@Getter`

File: app/src/main/java/cpe307/team6/toweroffense/game/ChibiCharacter.java [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+      return movingVectorX;
+   }
+
+   public int getMovingVectorY() {
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/ChibiCharacter.java]: this should be generated by lombok's `@Getter`

File: app/src/main/java/cpe307/team6/toweroffense/game/GameThread.java [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+            canvas = this.surfaceHolder.lockCanvas();
+
+            // Synchronized
+            synchronized (canvas) {
+               this.gameSurface.update();
+               this.gameSurface.draw(canvas);
+            }
+         } catch (Exception eException) {
+            // Do nothing.
+            System.out.println("GameThread: " + eException);
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/GameThread.java]: Android uses `Android.util.Log`

File: app/src/main/java/cpe307/team6/toweroffense/game/GameThread.java [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+         } catch (InterruptedException eException) {
+            System.out.println(eException);
+         }
+
+         startTime = System.nanoTime();
+         System.out.print(".");
+      }
+   }
+
+   void setRunning(final boolean running) {
```
Comment (thewells1024) [inline app/src/main/java/cpe307/team6/toweroffense/game/GameThread.java]: This should be generated by Lombok's `@Setter`

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 5 above)
Comment (calebhskim) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: These are used in the Character classes to draw onto the canvas.

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 6 above)
Comment (calebhskim) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: These are used in the Character classes to draw onto the canvas.

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 7 above)
Comment (calebhskim) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: These are used in the Character classes to draw onto the canvas.

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 8 above)
Comment (calebhskim) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: These are used in the Character classes to draw onto the canvas.

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 9 above)
Comment (calebhskim) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: These are used in the Character classes to determine how to break up the sprites.

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 10 above)
Comment (calebhskim) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: These are used in the Character classes to determine how to break up the sprites.

Comment (thewells1024) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: but they should be private with getters

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 9 above)
Comment (thewells1024) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: but they should be private with getters

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 8 above)
Comment (thewells1024) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: but they should be private with getters

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 7 above)
Comment (thewells1024) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: but they should be private with getters

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 6 above)
Comment (thewells1024) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: but they should be private with getters

File: app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java (hunk 5 above)
Comment (thewells1024) [reply app/src/main/java/cpe307/team6/toweroffense/game/AbstractGameObject.java]: but they should be private with getters

**Comment by thewells1024** [comment]
I'm gonna merge this, and I'll do a giant style fix and start working on gluing everything together

**Pull request merged**